lib/ui/deduplicate_page.dart

```diff
@@ -0,0 +1,378 @@
+
+  SortKey sortKey = SortKey.size;
+
+  @override
+  void initState() {
+    super.initState();
+    for (final duplicate in widget.duplicates) {
+      for (int index = 0; index < duplicate.files.length; index++) {
+        // Select all items but the first
+        if (index != 0) {
+          _selectedFiles.add(duplicate.files[index]);
+        }
+        // Maintain a map of fileID to fileSize for quick "space freed" computation
+        _fileSizeMap[duplicate.files[index].uploadedFileID] = duplicate.size;
+      }
+    }
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    _sortDuplicates();
+    return Scaffold(
+      appBar: AppBar(
+        title: Hero(
+          tag: "deduplicate",
+          child: Material(
+            type: MaterialType.transparency,
+            child: Text(
+              "deduplicate files",
+              style: TextStyle(
+                fontSize: 18,
+              ),
+            ),
+          ),
+        ),
+      ),
+      body: _getBody(),
+    );
+  }
+
+  void _sortDuplicates() {
+    widget.duplicates.sort((first, second) {
+      if (sortKey == SortKey.size) {
+        final aSize = first.files.length * first.size;
+        final bSize = second.files.length * second.size;
+        return bSize - aSize;
+      } else if (sortKey == SortKey.count) {
+        return second.files.length - first.files.length;
+      } else {
+        return second.files.first.creationTime - first.files.first.creationTime;
+      }
+    });
+  }
+
+  Widget _getBody() {
+    return Column(
+      mainAxisAlignment: MainAxisAlignment.center,
+      crossAxisAlignment: CrossAxisAlignment.center,
+      children: [
+        Expanded(
+          child: ListView.builder(
+            itemBuilder: (context, index) {
+              if (index == 0) {
+                return _getHeader();
+              } else if (index == 1) {
+                return _getSortMenu();
+              }
+              return Padding(
+                padding: const EdgeInsets.only(top: 10, bottom: 10),
+                child: _getGridView(widget.duplicates[index - kHeaderRowCount],
+                    index - kHeaderRowCount),
+              );
+            },
+            itemCount: widget.duplicates.length + kHeaderRowCount,
+            shrinkWrap: true,
+          ),
+        ),
+        _selectedFiles.isEmpty ? Container() : _getDeleteButton(),
+      ],
+    );
+  }
+
+  Padding _getHeader() {
+    return Padding(
+      padding: EdgeInsets.fromLTRB(16, 12, 16, 8),
+      child: Column(
+        children: [
+          Text(
+            "the following files were clubbed based on their sizes and timestamps",
+            style: TextStyle(
+              color: Colors.white.withOpacity(0.6),
+              height: 1.2,
+            ),
+          ),
+          Padding(
+            padding: EdgeInsets.all(4),
+          ),
+          Text(
+            "please review and delete the items you believe are duplicates",
```

Comment:
reminder about mentioning `long press` to review media.

Reply:
Have added a toast for now. Not too happy about it, will rethink during UI revamp.